### PR TITLE
DOC for migrating from EEGLAB

### PR DESCRIPTION
**Pull request by jasmainak**
Taking over https://github.com/mne-tools/mne-python/pull/2746/ and keeping it simple stupid

Ready to merge from my end. Here's a screenshot.

migrating

### CODE REVIEW

File: doc/manual/migrating.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. _migrating
+
+Migrating from EEGLAB
+=====================
+
+To read in data exported from EEGLAB, MNE offers :ref:`EDF and set file readers <ch_convert>`.
+
+Here is a cheatsheet to help users migrate painlessly from EEGLAB. For the sake of clarity, let us assume
+that the following are already defined or known: the file name ``fname``, low pass cut off frequency ``lfreq``,
```
Comment (kingjr) [inline doc/manual/migrating.rst]: low-pass cut-off

**Comment by kingjr** [comment]
otherwise LGTM

**Comment by Eric89GXL** [comment]
Looks great!

For the last entry in the right column, can that be multi line instead of
being separated by "or"?

Probably should have the object result like "raw = mne.io.Raw (..." in the
first one.

**Comment by Eric89GXL** [comment]
Also, it should link to the doc for those commands if possible

**Comment by Eric89GXL** [comment]
I wonder if there is a way to get e.g. the raw.save line to link to
Raw.save.

**Comment by teonlamont** [comment]
should we note that the importing that we list as an example is for the `.fif` and that you would need to use system-specific format for reading?

**Comment by jasmainak** [comment]
@Eric89GXL no luck with linking the methods. But the multiline table works

**Comment by kingjr** [comment]
> should we note that the importing that we list as an example is for the .fif and that you would need to use system-specific format for reading?

+1 . If there's a doc page that explains format reading, I'd add a `see <ref> for other file format`

**Comment by jasmainak** [comment]
comments addressed. updated page looks like this
migrating

**Comment by Eric89GXL** [comment]
We should probably list read_raw_edf and read_raw_eeglab. Multi line looks
good.

**Comment by mmagnuski** [comment]
What do you think about being a little more explicit about the imports? I mean instead of `io.Raw(...)` writing `mne.io.Raw` or adding somewhere `from mne import io`. This might be helpful for beginners coming from matlab - a land where no imports and no real namespaces exist.

**Comment by jona-sassenhagen** [comment]
Great job, I hope I'll have time to review later.

**Comment by Eric89GXL** [comment]
@jasmainak added links:

https://github.com/jasmainak/mne-python/pull/1

**Comment by Eric89GXL** [comment]
(PR also gives examples of more explicit imports like @mmagnuski mentioned)

**Comment by agramfort** [comment]
thanks guys for taking the time ! that's very useful.

**Comment by jona-sassenhagen** [comment]
Can the code be shown as code? I mean, highlight it as fixed-width etc.

>What do you think about being a little more explicit about the imports? I mean instead of io.Raw(...) writing mne.io.Raw or adding somewhere from mne import io. This might be helpful for beginners coming from matlab - a land where no imports and no real namespaces exist.

I emphatically agree. This must somehow be made clear.

File: doc/manual/migrating.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+||                 ||                                                            || evoked.save(fname)                                |
++------------------+-------------------------------------------------------------+----------------------------------------------------+
+
+Note that MNE has functions to read a variety of file formats, not just :func:`mne.io.Raw`. The interested user is directed to the :ref:`IO documentation <ch_convert>`.
+
+Pitfalls
+--------
+
+* Python operations are natively in place. To explicitly ask for a copy like in Matlab, use the boolean argument ``copy`` provided in MNE functions. 
+* The concept of channel types is critical in MNE because it supports analysis of multimodal data (e.g., EEG, MEG, EOG) whereas most EEGLAB
```
Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: >```(e.g., EEG, MEG, EOG) ```
Add stim channels, these are the most confusing.

File: doc/manual/migrating.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+|| Save data       || EEG = pop_saveset(EEG, fname);                             || epochs.save(fname)                                |
+||                 ||                                                            || evoked.save(fname)                                |
++------------------+-------------------------------------------------------------+----------------------------------------------------+
+
+Note that MNE has functions to read a variety of file formats, not just :func:`mne.io.Raw`. The interested user is directed to the :ref:`IO documentation <ch_convert>`.
+
+Pitfalls
+--------
+
+* Python operations are natively in place. To explicitly ask for a copy like in Matlab, use the boolean argument ``copy`` provided in MNE functions.
```
Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: Operations? You mean methods? A lot of our functions are not in place (for to us, but probably not a beginner, obvious reasons). Reformulate.

File: doc/manual/migrating.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+Migrating from EEGLAB
+=====================
+
+To read in data exported from EEGLAB, MNE offers an EDF reader :func:`mne.io.read_raw_edf` and a `set` file reader.
+To read in `set` files containing ``raw`` data, use :func:`mne.io.read_raw_eeglab` and to read in `set` files containing
+``epochs`` data, use :func:`mne.read_epochs_eeglab`.
+
+Here is a cheatsheet to help users migrate painlessly from EEGLAB. For the sake of clarity, let us assume
+that the following are already defined or known: the file name ``fname``, low-pass cut-off frequency ``lfreq``,
+time interval of the epochs ``tmin`` and ``tmax``, and the conditions ``cond1`` and ``cond2``.
```
Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: This reads amateurish (style wise).

Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: Make it explicit?

```
from mne import io
lfreq, hfreq = 1, 30

fname = "/.../"
cond1, cond2 = ...
```

And we don't necessarily need tmin and tmax anymore, with the most recent changes, don't we?

**Comment by jasmainak** [comment]
@Eric89GXL thanks for the PR. I've addressed comments by @jona-sassenhagen

File: doc/manual/migrating.rst [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
++-------------------+--------------------------------------------------------------+-----------------------------------------------------------------------------+
+| Processing step   | EEGLAB function                                              | MNE                                                                         |
++===================+==============================================================+=============================================================================+
+| Get started       | addpath('EEGLAB')                                            | | import mne                                                                |
```
Comment (mmagnuski) [inline doc/manual/migrating.rst]: to get started with eeglab one would need (once the path to main eeglab dir is added, which can be done only once and saved for future use) to open eeglab inteface with `eeglab` command - this adds all other paths needed but also litters the workspace with multiple variables. Another way would be to:
```
addpath(genpath(fileparts(which('eeglab'))));
```
(but this is rather unweildy and adds too many paths)

Comment (Eric89GXL) [inline doc/manual/migrating.rst]: So would this be good enough:
```
addpath(...)
eeglab
```
I don't think it's critical we get this part ideal and perfect, just show
something eeglab people will understand as "getting set up".

Comment (mmagnuski) [inline doc/manual/migrating.rst]: Definitely - that's good.

Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: I think the most ecologically valid would be just 
```
eeglab;
```

Cause people usually just put EEGLAB in their path IME.

And that could sit parallel to

```
import mne
...
```
etc

Comment (jasmainak) [inline doc/manual/migrating.rst]: what is the consensus on this?

Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: I'm okay with `addpath`, if Eric and @mmagnuski prefer it.

Comment (jona-sassenhagen) [inline doc/manual/migrating.rst]: I'm okay with `addpath`, if Eric and @mmagnuski prefer it.

File: doc/manual/migrating.rst [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+Migrating from EEGLAB
+=====================
+
+To read in data exported from EEGLAB, MNE offers an EDF reader :func:`mne.io.read_raw_edf` and a `set` file reader.
+To read in `set` files containing ``raw`` data, use :func:`mne.io.read_raw_eeglab` and to read in `set` files containing
+``epochs`` data, use :func:`mne.read_epochs_eeglab`.
+
+Here is a cheatsheet to help users migrate painlessly from EEGLAB. For the sake of clarity, let us assume
+that the following are already defined or known: the file name ``fname``, low-pass cut-off frequency ``lfreq``,
```
Comment (agramfort) [inline doc/manual/migrating.rst]: low-pass cut-off frequency ``lfreq`` -> Low cut-off frequency in Hz ``l_freq``

the l_freq sets the cut-off for the high pass filter.

File: doc/manual/migrating.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+|                   |                                                              | | from mne.preprocessing import     :class:`ICA <mne.preprocessing.ICA>`    |
++-------------------+--------------------------------------------------------------+-----------------------------------------------------------------------------+
+| Import data       | EEG = pop_fileio(fname);                                     | | :func:`raw = io.Raw(fname) <mne.io.Raw>`                                  |
+|                   |                                                              | | :func:`raw = io.read_raw_edf(fname) <mne.io.read_raw_edf>`                |
+|                   |                                                              | | :func:`raw = io.read_raw_eeglab(fname) <mne.io.read_raw_eeglab>`          |
++-------------------+--------------------------------------------------------------+-----------------------------------------------------------------------------+
+| Filter data       | EEG = pop_eegfiltnew(EEG, lfreq, hfreq);                     | :func:`raw.filter(lfreq, hfreq) <mne.io.Raw.filter>`                        |
```
Comment (agramfort) [inline doc/manual/migrating.rst]: lfreq, hfreq -> l_freq, h_freq

**Comment by agramfort** [comment]
besides +1 for merge

thx @jasmainak

**Comment by jasmainak** [comment]
comments addressed @agramfort

**Comment by Eric89GXL** [comment]
Thanks @jasmainak

**Pull request merged**